GAE/src/org/waterforpeople/mapping/dataexport/GraphicalSurveySummaryExporter.java

```diff
@@ -330,38 +330,40 @@
     public void export(Map<String, String> criteria, File fileName,
             String serverBase, Map<String, String> options) {
         final String surveyId = criteria.get(SurveyRestRequest.SURVEY_ID_PARAM).trim();
+        final String apiKey = criteria.get("apiKey").trim();
 
         processOptions(options);
 
         questionsById = new HashMap<Long, QuestionDto>();
-        surveyGroupDto = BulkDataServiceClient.fetchSurveyGroup(surveyId, serverBase,
-                criteria.get("apiKey").trim());
         this.serverBase = serverBase;
         boolean useQuestionId = "true".equals(options.get("useQuestionId"));
         String from = options.get("from");
         String to = options.get("to");
         String limit = options.get("maxDataReportRows");
         try {
-            Map<QuestionGroupDto, List<QuestionDto>> questionMap = loadAllQuestions(surveyId,
-                    performGeoRollup, serverBase, criteria.get("apiKey"));
-            if (questionMap != null) {
+            Map<QuestionGroupDto, List<QuestionDto>> questionMap = 
+                    loadAllQuestions(surveyId, performGeoRollup, serverBase, apiKey);
```

Comment:
Could we already pass the `useQuestionId` parameter here and if its `true` then we already have the question options loaded in this call?